lib/index.js

```diff
@@ -34,6 +34,8 @@ var CoverallsReporter = function(rootConfig, helper, logger) {
         filepath = path.resolve(rootConfig.basePath, reporter.dir);
         break;
       }
+    } else {
+     throw new Error("coverage reporter should precede coveralls");
```

Comment:
Basically, the `coverage` reporter needs to fire first, in order to ensure that the `lcov` output is emitted at all at the time this reporter runs.

Above, the following code catches the case where `coverage` isn't used at all, or doesn't precede coveralls:

```js
   if (coverage < 0 || coverage > reporters.indexOf('coveralls')) {
     throw new Error("coverage reporter should precede coveralls");
   }
```

So I was wrong before, this isn't doing what I thought it was doing, and isn't really needed

---

> What if the user is using another type? The plugin would still try to find the file, wouldnt it?

Sure. Any other type is not necessarily bad. If the default configuration is used, there's no type, and it defaults to `html`, which produces an lcov.info.

If it can't find the file, it tries again a small number of times, with a timeout, to work around a race condition that had been observed previously.

It should probably log a message on the 5th retry failure, just so that you can figure out what's going on.